### PR TITLE
Pass project name as --prompt for virtualenv

### PR DESCRIPTION
**Pull request by Tadaboody**
# Pull Request Check List

Resolves: #3055

- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

### CODE REVIEW

**Comment by Tadaboody** [comment]
@abn I think this is ready for review, how do we start the process?

**Comment by Tadaboody** [comment]
@abn Bumping again for review, also would appreciate a `hacktoberfest-accepted` tag if you deem this PR worthy

**Comment by abn** [comment]
Definitely worthy 😉 Just haven't managed to get the time to get back to this. Will do so in the next day or so.

File: docs/docs/configuration.md [hunk 1]
```diff
@@ -128,6 +129,12 @@ If not set explicitly (default), `poetry` will use the virtualenv from the `.ven
 Directory where virtual environments will be created.
 Defaults to `{cache-dir}/virtualenvs` (`{cache-dir}\virtualenvs` on Windows).
 
+### `virtualenvs.prompt` : string
+
+Name of the command line prompt to be displayed when the virtual enviroment is activated.
+The variables `project_name` and `python_version` are avalible for formatting.
```
Comment (stephsamson) [inline docs/docs/configuration.md:135]:
```suggestion
The variables `project_name` and `python_version` are available for formatting.
```

File: docs/docs/configuration.md [hunk 2]
```diff
@@ -128,6 +129,12 @@ If not set explicitly (default), `poetry` will use the virtualenv from the `.ven
...
+Defaults to "{project_name}-py{python_version}"
```
Comment (stephsamson) [inline docs/docs/configuration.md:136]:
```suggestion
Defaults to "{project_name}-py{python_version}".
```

File: poetry/config/config.py [hunk 3]
```diff
@@ -128,7 +129,11 @@ def process(self, value):  # type: (Any) -> Any
         if not isinstance(value, basestring):
             return value
 
-        return re.sub(r"{(.+?)}", lambda m: self.get(m.group(1)), value)
+        return re.sub(
+            r"{(.+?)}",
+            lambda m: self.get(m.group(1)) or "{{{key}}}".format(key=m.group(1)),
```
Comment (stephsamson) [inline poetry/config/config.py:134]: This lambda is getting difficult to read at first pass. I'd suggest to take encapsulate this into its own function.

**Review by stephsamson — CHANGES_REQUESTED** [review]

**Comment by Tadaboody** [comment]
Hey, I've been fixing merge conflicts here and there and would love to get to merge this before it gets stale again, any chance this gets looked at?

**Comment by nikhilweee** [comment]
I would love to get this merged! @abn @stephsamson Please review this as soon as possible?

File: poetry/config/config.py (hunk 3 above)
Comment (Tadaboody) [reply poetry/config/config.py]: I think I extracted it well enough [here](https://github.com/python-poetry/poetry/pull/3057/files#diff-302fd6e399fc926506961839561ed7635a346a4792af1d47d2cfa07c5651b7fbR138), should I change something? I'd still like to get this merged (and stop worrying about conflicts with upstream)

**Comment by abn** [comment]
@can I get this rebased please? Aologies for the delay in getting to this.

**Comment by neersighted** [comment]
Superseded by #5606.

**Pull request closed without merging**